lib/rules/function-calc-no-unspaced-operator/index.mjs

```diff
@@ -66,167 +71,171 @@ const rule = (primary, _secondaryOptions, context) => {
+
+				if (!OPERATOR_REGEX.test(operand)) return;
+
+				const beforeOperandNode = parent.value[index - 1];
+				const isFirstOperand = !parent.value.slice(0, index).some(isTokenNode);
+				const isLastOperand = !parent.value.slice(index + 1).some(isTokenNode);
+				const hasOperatorBeforeOperand = parent.value
+					.slice(0, index)
+					.findLast((node) => isOperatorNode(node, ALL_OPERATORS));
+
+				let fixedOperand = operand;
+				let fixedCount = 0;
+
+				// Scan operator characters in an operand token
+				for (const matched of operand.matchAll(new RegExp(OPERATOR_REGEX, 'g'))) {
+					const operator = matched[0];
+					const operatorIndex = matched.index;
+
+					// Ignore a number sign in the first operand, e.g. "calc(-1px)"
+					if (operatorIndex === 0 && isFirstOperand) continue;
+
+					// Ignore a suffixed operator in the last operand, e.g. "calc(1px-)"
+					if (operatorIndex === operand.length - 1 && isLastOperand) continue;
+
+					const beforeOperator = operand.charAt(operatorIndex - 1);
+
+					if ((beforeOperator && beforeOperator !== ' ') || isOperandNode(beforeOperandNode)) {
+						if (context.fix) {
+							fixedOperand = insertCharAtIndex(fixedOperand, operatorIndex + fixedCount, ' ');
+							fixedCount++;
+						} else {
+							complain(
+								messages.expectedBefore(operator),
+								decl,
+								valueIndex + operandIndex + operatorIndex,
+								operator,
+							);
+						}
+					}
 
-				// E.g. "10px * -2" when the last node is "-2"
-				if (isOperator(nodes.at(-3), ALL_OPERATORS) && isSingleSpace(nodes.at(-2))) {
-					return false;
+					const afterOperator = operand.charAt(operatorIndex + 1);
+
+					if (afterOperator && afterOperator !== ' ' && !hasOperatorBeforeOperand) {
+						if (context.fix) {
+							fixedOperand = insertCharAtIndex(fixedOperand, operatorIndex + fixedCount + 1, ' ');
+							fixedCount++;
+						} else {
+							complain(
+								messages.expectedAfter(operator),
+								decl,
+								valueIndex + operandIndex + operatorIndex,
+								operator,
+							);
+						}
+					}
 				}
 
-				if (context.fix) {
+				if (fixedCount > 0) {
 					needsFix = true;
-					lastNode.value = insertCharAtIndex(lastNode.value, operatorIndex + 1, ' ').trim();
-					lastNode.value = insertCharAtIndex(lastNode.value, operatorIndex, ' ').trim();
-
-					return true;
+					operandToken[1] = fixedOperand;
 				}
-
-				const operator = lastNode.value.charAt(operatorIndex);
-
-				complain(
-					messages.expectedOperatorBeforeSign(operator),
-					decl,
-					valueIndex + lastNode.sourceIndex + operatorIndex,
-					operator,
-				);
-
-				return true;
 			}
 
-			const parsedValue = valueParser(value);
+			const nodes = parseListOfComponentValues(tokenize({ css: value }));
 
-			parsedValue.walk((node) => {
-				if (!isFunction(node) || !FUNC_NAMES_REGEX.test(node.value)) return;
+			if (nodes.length === 0) return;
 
-				const { nodes } = node;
+			walk(nodes, ({ node: funcNode }) => {
```

Comment:
Should we add beter recursion support?

```
calc((1px+2px) + 3px)
```

The `(1px+2px)` simple block is also a `calc-sum` and should be detected as an error.

This is currently also ignored : https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLjiADvCmEgDZgAUlAjAQB4DUATAwJQqMoDM7A3FgC+IADQgAZgEsyMAHJIAtnEQx6igjIB0YAM47R4CJikBzBCAyYUKdCFU4YmKDtvEA2lmvXbOnAE8ZMklMHABaSGNJE1DfJCckACcoW08UAF0hAwjTADEIBIUkHHMAKx0jA1gCfURLbxBfAJggkNcbEDIimF9bEVSff0DgsOyomJw4qETkhHbOhx6QTMEgA

Adding support for this is relatively simple with the walker states.